File: sdks/php/rest/src/Types/ApiKeyListItem.php
<?php

namespace Brume\Rest\Types;

use Brume\Rest\Core\Json\JsonSerializableType;
use Brume\Rest\Core\Json\JsonProperty;
use Brume\Rest\Core\Types\ArrayType;

class ApiKeyListItem extends JsonSerializableType
{
    /**
     * @var string $environment
     */
    #[JsonProperty('environment')]
    public string $environment;

    /**
     * @var string $id
     */
    #[JsonProperty('id')]
    public string $id;

    /**
     * @var string $label
     */
    #[JsonProperty('label')]
    public string $label;

    /**
     * @var ?string $lastUsedAt RFC-3339 timestamp of last use, or `null` if never used.
     */
    #[JsonProperty('last_used_at')]
    public ?string $lastUsedAt;

    /**
     * @var string $prefix
     */
    #[JsonProperty('prefix')]
    public string $prefix;

    /**
     * @var array<value-of<ApiKeyScope>> $scopes Granted scopes (E6). Serializes as snake_case strings.
     */
    #[JsonProperty('scopes'), ArrayType(['string'])]
    public array $scopes;

    /**
     * @param array{
     *   environment: string,
     *   id: string,
     *   label: string,
     *   prefix: string,
     *   scopes: array<value-of<ApiKeyScope>>,
     *   lastUsedAt?: ?string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->environment = $values['environment'];
        $this->id = $values['id'];
        $this->label = $values['label'];
        $this->lastUsedAt = $values['lastUsedAt'] ?? null;
        $this->prefix = $values['prefix'];
        $this->scopes = $values['scopes'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: sdks/csharp/rest/src/Brume.Rest/Types/ApiKeyListItem.cs
using Brume.Rest.Core;
using global::System.Text.Json;
using global::System.Text.Json.Serialization;

namespace Brume.Rest;

[Serializable]
public record ApiKeyListItem : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("environment")]
    public required string Environment { get; set; }

    [JsonPropertyName("id")]
    public required string Id { get; set; }

    [JsonPropertyName("label")]
    public required string Label { get; set; }

    /// <summary>
    /// RFC-3339 timestamp of last use, or `null` if never used.
    /// </summary>
    [JsonPropertyName("last_used_at")]
    public string? LastUsedAt { get; set; }

    [JsonPropertyName("prefix")]
    public required string Prefix { get; set; }

    /// <summary>
    /// Granted scopes (E6). Serializes as snake_case strings.
    /// </summary>
    [JsonPropertyName("scopes")]
    public IEnumerable<ApiKeyScope> Scopes { get; set; } = new List<ApiKeyScope>();

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
